### PR TITLE
UID2-2330 Add workflow to build and publish pypi packages

### PR DESCRIPTION
**Pull request by cYKatherine**
PRs:
- https://github.com/IABTechLab/uid2-shared-actions/pull/81
- https://github.com/IABTechLab/uid2-client-python/pull/26

Successful run: https://github.com/IABTechLab/uid2-client-python/actions/runs/8228106898/job/22497018541
Successful release: https://pypi.org/project/uid2-client/2.2.3/#history
Draft release created on github: https://github.com/IABTechLab/uid2-client-python/releases

### CODE REVIEW

File: .github/workflows/build-and-publish.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        - Minor
+        - Patch
+      publish_vulnerabilities:
+        type: string
+        default: true
+
+jobs:
+  build-and-pubish:
+    name: Build and publish Python packages to PyPi
+    uses: IABTechLab/uid2-shared-actions/.github/workflows/shared-publish-to-pypi-versioned.yaml@kcc-UID2-2330-create-workflow-for-python
```
Comment (cYKatherine) [inline .github/workflows/build-and-publish.yaml:20]: Will revert to `v2` before merging

**Comment by sunnywu** [comment]
are you gonna delete existing publish.yaml?

File: .github/workflows/build-and-publish.yaml (hunk 1 above)
Comment (thomasm-ttd) [inline .github/workflows/build-and-publish.yaml:20]: Should you not also pass through the publish_vulnerability value?

Comment (cYKatherine) [reply .github/workflows/build-and-publish.yaml:24]: Added

**Pull request merged**